flux-kvs(1): improve man page formatting

garlick opened the pull request:
Problem: flux-kvs(1) is hard to use as a reference.

Improve the synopsis, description, and break out sub-commands into sub-sections with properly declared options.

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+
+.. option:: -A, --append
+
+  With :option:`--watch`, display only the new data when the watched
+  is appended to.
```
"when the watched **value** is appended to"?

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+.. program:: flux kvs put
+
+Set *key* to *value*.  If *key* exists, the current value is overwritten.
+If multiple *key=value* pairs are specified, they are send as one
```
s/send/sent/

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+the value is zero length.
+
+
+.. option:: -h, --help
+
+  Display help on this sub-command.
+
+.. option:: -N, --namespace=NAME
```
personal preference ... `--help` option feels the least useful in the manpage (i.e. you're already in the manpage).  List it last amongst all the options?

garlick replied:
Great point. Instead of repeating that option and the namespace one, I'll just mention it in the section intro.

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+
+.. option:: -A, --append
+
+   Append value(s) to key instead of overwriting.
```
no `(s)`?  Or should be `key(s)`?  I'd leave it singular.

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+
+.. program:: flux kvs readlink
+
+Print the symbolic link target of *key*.  The target may be another key name,
+or a :option:`namespace::key` tuple.
```
similar to other sections above, say it's an error if `key` is not a link?

chu11 commented on doc/man1/flux-kvs.rst:
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
+  Initialize namespace with specific root directory reference
+  If unspecified, an empty directory is referenced.
+
+namespace remove
+----------------
+
+.. program:: flux kvs namespace remove
+
+Remove a KVS namespace.
+
+.. option:: -h, --help
+
+  Display help on this sub-command.
```
perhaps note that remove is done in the background? i.e. may not be removed when command returns.

chu11 reviewed (commented):
overall looks good, just some small comments

garlick commented:
Thanks @chu11, that was super helpful.  I fixed those things and also reworded the last paragraph of the DESCRIPTION about namespaces, and added RFC 18 to the FLUX RFC section.

chu11 reviewed (approved):
LGTM
